### PR TITLE
Task 4 components part 2

### PR DESCRIPTION
**Pull request by OlehKhozhai**

### CODE REVIEW

File: src/components/MoviesNavbarSortBy/index.tsx [hunk 1]
```diff
@@ -2,17 +2,33 @@ import React from 'react';
 import cn from 'classnames';
 
 import Dropdown from 'components/_common/Dropdown';
+import useDropdown from 'hooks/useDropdown';
 import styles from './styles.module.scss';
 
+const options = ['release date', 'rating', 'best 100'];
+
```
Comment (yurii-hasiuk) [inline src/components/MoviesNavbarSortBy/index.tsx:9]: I would keep such kind of options in some config.  But it's up to you and not necessary for now

File: src/components/_common/Dropdown/index.tsx [hunk 2]
```diff
@@ -1,41 +1,48 @@
...
+  useClickOutside({ ref: dropdownRef, callback: onClose });
 
   return (
-    <div className={cn(styles.root, className)}>
-      <span className={styles.value} onClick={handleOpenDropdownClick}>
-        {value}
-      </span>
-
+    <div className={cn(styles.root, className)} ref={dropdownRef}>
+      {value ? (
```
Comment (yurii-hasiuk) [inline src/components/_common/Dropdown/index.tsx:33]: I think we can simplify this code

` <span className={ cn ( value ? styles.value : styles.threeDots) } onClick={onToggle}>
          {value || '...'}
  </span>
`

File: src/components/_common/FormFields/Input/index.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+type InputProps = {
+  label: string;
+  name: string;
+  className?: string;
+} & React.InputHTMLAttributes<HTMLInputElement>;
+
+const Input: React.FC<InputProps> = ({ label, ...props }) => {
+  const [showPassword, setShowPassword] = React.useState(false);
+  const [field, meta] = useField({ ...props });
+  const isPasswordType = props.type === 'password';
```
Comment (yurii-hasiuk) [inline src/components/_common/FormFields/Input/index.tsx:17]: Can we use destructuring from `props` for `type` too?

File: src/components/_common/Modal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    return () => {
+      document.body.style.overflowY = 'auto';
+    };
+  }, [isOpen]);
+
+  if (!modalNode || !isOpen) {
+    return null;
+  }
+
+  return createPortal(component, modalNode);
```
Comment (yurii-hasiuk) [inline src/components/_common/Modal/index.tsx:41]: I would keep one `return` statement

**Review by yurii-hasiuk — CHANGES_REQUESTED** [review]

File: src/components/MoviesNavbarSortBy/index.tsx (hunk 1 above)
Comment (OlehKhozhai) [reply src/components/MoviesNavbarSortBy/index.tsx:9]: Yep, good point, I will move in the next task.

File: src/components/_common/Dropdown/index.tsx (hunk 2 above)
Comment (OlehKhozhai) [reply src/components/_common/Dropdown/index.tsx:33]: make sense, will fix in the next task

File: src/components/_common/FormFields/Input/index.tsx (hunk 3 above)
Comment (OlehKhozhai) [reply src/components/_common/FormFields/Input/index.tsx:17]: make sense, will fix in the next task

File: src/components/_common/Modal/index.tsx (hunk 4 above)
Comment (OlehKhozhai) [reply src/components/_common/Modal/index.tsx:41]: make sense, will fix in the next task

**Pull request merged**